pkg/deploy/lattice/service_manager.go

```diff
@@ -204,29 +204,28 @@ func (m *defaultServiceManager) getAllAssociations(ctx context.Context, svcSum *
 func (m *defaultServiceManager) updateAssociations(ctx context.Context, svc *Service, svcSum *SvcSummary) error {
 	assocs, err := m.getAllAssociations(ctx, svcSum)
 	if err != nil {
-		return err
+		return fmt.Errorf("in updateAssociations, getAllAssociations failed with %w", err)
```

Comment:
adding a some more detailed error messages since the err here is just

```status code: 404, request id: ... ```

Reply:
Idiomatic error string would be ```"update associations: get all associations: %w"``` lower case for function names, chained by colon. Also it's better to use one of the approaches for error composition:

1. function that throw error need to add it's own name to errors
2. or caller adds names of functions it calls

Otherwise you might have duplicates in the final error message. Here both are mixed, updateAssociation adds itself and getAllAssociations. You probably want to move getAllAssociations part into it's own function. For example:
in updateAssociations:
```go
return fmt.Errorf("update associations: %w, err);
```
in getAllAssociations
```go
return fmt.Errorf("get all associations: %w, err);
```

Also dont use words "failed" if it's not original error, there is a tendency to add "failed", "unsuccessful", etc when we wrap errors, so when error pop-up to the top level of stack it will have 5-6 "failed" words in single sentence. So back to the first comment, chain method names with colons, and actual error last in the chain.